http-clients/src/main/java/com/palantir/remoting/http/UserAgentInterceptor.java

```diff
@@ -40,8 +41,8 @@ public static UserAgentInterceptor of(String userAgent) {
 
     @Override
     public void apply(RequestTemplate template) {
-        if (!template.headers().containsKey("User-Agent")) {
-            template.header("User-Agent", userAgent);
+        if (!HeaderAccessUtils.caseInsensitiveContains(template.headers(), HttpHeaders.USER_AGENT)) {
```

Comment:
to expand on that, the test we could do for `caseInsensitiveContains` should do the loop but not the map creation.